apps/meteor/app/federation-v2/server/index.ts

```diff
@@ -29,7 +33,11 @@ export const federationRoomServiceSender = FederationFactory.buildRoomServiceSen
 
 export const runFederation = async (): Promise<void> => {
 	queueInstance.setHandler(federationEventsHandler.handleEvent.bind(federationEventsHandler), FEDERATION_PROCESSING_CONCURRENCY);
+
 	await federation.start();
+
+	rocketChatRoomCallbacksHandler.injectBridgeInstance(federation);
```

Comment:
In my opinion, it would be better to have a method on the Factory to set up all the listeners and from them, we call the appropriate handler.
Something like: 

```
setupListener(roomServiceSender: FederationRoomServiceSender): void {
  RocketChatRoomCallbacksHandler.onRoomLeft(async (user: IUser, room: IRoom) => await roomServiceSender.onUserLeftRoom(FederationRoomSenderConverter.convert(roomServiceSender.onRoomLeft(FederationRoomSenderConverter.convert()))));
}
```
And so we could move the business logic and centralize it to the service. 

```
public async onRoomLeft(inputDto: Dto): Promise<void> {
 await this.bridge.leaveRoom(rocketRoom._id, rocketUser._id);
}
```